lib/restclient/payload.rb

```diff
@@ -116,9 +116,12 @@ def inspect
       end
 
       def short_inspect
-        (size > 500 ? "#{size} byte(s) length" : inspect)
+        if RestClient.limit_payload_log? && size > 500
+          "#{size} byte(s) length"
+        else
+          inspect
```

Comment:
The sensitive information could have been logged without removing the limitation, so I don't see a reason for a special warning. We're not changing the current situation.